code/iaas/storage-service/src/main/java/io/cattle/platform/storage/api/filter/ExternalTemplateInstanceFilter.java

```diff
@@ -79,51 +83,33 @@ protected static void validateImageUuid(String image) {
         }
 
         String server = dockerImage.getServer();
-        String qualifiedName = dockerImage.getQualifiedName();
-        String whitelist = WHITELIST_REGISTRIES.get();
+        List<String> whitelistRegistries = WHITELIST_REGISTRIES.get();
         String defaultRegistry = DEFAULT_REGISTRY.get();
         String userProvidedRegistry = dockerImage.getServer();
 
-        if (!qualifiedName.startsWith(server)) {
+        if (!image.contains(server)) {
```

Comment:
The qualifiedName field is a field from dockerImage, that is created by DockerImage.Parse
In the case when user explicitly prepended `index.docker.io` to the image, DockerImage.Parse was stripping `index.docker.io` off the image name, and then the validateImage function would never get to see that user had provided this registry. It would fail if the default registry had been set to something else. Say `quay.io` because in the absence of userProvidedRegistry, I'm prepending the default registry to image